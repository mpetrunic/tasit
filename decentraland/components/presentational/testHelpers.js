import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: {
    type: ESTATE,
    id: "123",
    name: "Sample Estate",
    img: "https://api.decentraland.org/v1/estates/1/map.png",
  },
};

export const parcelForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: {
    type: PARCEL,
    id: "0123456789",
    name: "Sample Parcel",
    img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
  },
};

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
};
