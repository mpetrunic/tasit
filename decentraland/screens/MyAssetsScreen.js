import React from "react";
import { connect } from "react-redux";
import { setMyAssetsList } from "../redux/actions";
import PropTypes from "prop-types";
import MyAssetsList from "@presentational/MyAssetsList";
import MyAssetsListItem from "@presentational/MyAssetsListItem";
import { listsAreEqual } from "@helpers";
import { getAssetsOf } from "@helpers/decentraland";
import { storeMyAssets } from "@helpers/storage";

export class MyAssetsScreen extends React.Component {
  componentDidMount = async () => {
    const { account, myAssets: fromState, setMyAssetsList } = this.props;
    if (account) {
      const { address } = account;
      const fromBlockchain = await this._getAssetsFromBlockchain(address);
      const shouldUpdate = !listsAreEqual(fromBlockchain, fromState);
      if (shouldUpdate) {
        setMyAssetsList(fromBlockchain);
        storeMyAssets(fromBlockchain);
      }
    }
  };

  _getAssetsFromBlockchain = async address => {
    const listOfPromises = await getAssetsOf(address);
    const fromBlockchain = await Promise.all([...listOfPromises]);
    return fromBlockchain;
  };

  _renderItem = ({ item: asset }) => {
    return <MyAssetsListItem asset={asset} />;
  };

  render() {
    const { myAssets } = this.props;

    return (
      <MyAssetsList myAssetsList={myAssets} renderItem={this._renderItem} />
    );
  }
}

MyAssetsScreen.propTypes = {
  myAssets: PropTypes.array.isRequired,
  account: PropTypes.object,
};

const mapStateToProps = state => {
  const { myAssets, accountInfo } = state;
  const { account } = accountInfo;
  const { list: myAssetsList } = myAssets;
  return { myAssets: myAssetsList, account };
};

const mapDispatchToProps = {
  setMyAssetsList,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MyAssetsScreen);
