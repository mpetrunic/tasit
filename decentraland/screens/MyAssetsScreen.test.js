import React from "react";
import { shallow } from "enzyme";
import { MyAssetsScreen } from "./MyAssetsScreen";
import { parcel } from "@presentational/testHelpers";

describe("ListLandForSaleScreen", () => {
  describe("renders the component", () => {
    it("without assets", async () => {
      const myAssets = { list: [] };
      expect(shallow(<MyAssetsScreen myAssets={myAssets} />)).toMatchSnapshot();
    });
    it("with assets", async () => {
      const myAssets = { list: [parcel] };
      expect(shallow(<MyAssetsScreen myAssets={myAssets} />)).toMatchSnapshot();
    });
  });
});
