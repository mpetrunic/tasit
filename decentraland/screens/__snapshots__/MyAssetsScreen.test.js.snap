// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLandForSaleScreen renders the component with assets 1`] = `
<MyAssetsList
  myAssetsList={
    Array [
      Object {
        "id": "0123456789",
        "img": "https://api.decentraland.org/v1/parcels/0/0/map.png",
        "name": "Sample Parcel",
        "type": "PARCEL",
      },
    ]
  }
  renderItem={[Function]}
/>
`;

exports[`ListLandForSaleScreen renders the component without assets 1`] = `
<MyAssetsList
  myAssetsList={Array []}
  renderItem={[Function]}
/>
`;
